### PR TITLE
Add functionality to remove bugs from a local repository.

### PR DESCRIPTION
**Pull request by Invincibot**
This adds a function to remove git references in the repo package and another one to remove bugs in the bug package.

### CODE REVIEW

File: bug/bug_test.go [hunk 1]
```diff
@@ -99,6 +100,12 @@ func TestBugCommitLoad(t *testing.T) {
 	bug3, err := ReadLocalBug(repo, bug1.Id())
 	assert.NoError(t, err)
 	equivalentBug(t, bug1, bug3)
+
+	err = RemoveLocalBug(repo, bug1.Id())
+	assert.NoError(t, err)
+
+	streamedBugs := ReadAllLocalBugs(repo)
+	assert.Equal(t, 0, len(streamedBugs))
```
Comment (MichaelMure) [inline bug/bug_test.go:108]: It's subtle, but there is a race here. `streamedBugs` is a channel. `len(streamedBugs)` return the number of element **at the moment** in the channel, but `ReadAllLocalBugs()` might not have started pushing values there yet.

To count the number or bugs, you need to really read the channel and count the values.

File: repository/repo_testing.go [hunk 2]
```diff
@@ -156,7 +156,7 @@ func RepoTest(t *testing.T, creator RepoCreator, cleaner RepoCleaner) {
 
 		commits, err := repo.ListCommits("refs/bugs/ref2")
 		require.NoError(t, err)
-		assert.Equal(t, []Hash{commit1, commit2}, commits)
+		assert.ElementsMatch(t, []git.Hash{commit1, commit2}, commits)
```
Comment (MichaelMure) [inline repository/repo_testing.go:159]: test is failing here, bad merge I suppose
```suggestion
		assert.ElementsMatch(t, []Hash{commit1, commit2}, commits)
```

**Comment by pawamoy** [comment]
So it's a partial implementation for #66?

**Comment by MichaelMure** [comment]
Yes

**Comment by MichaelMure** [comment]
:+1:

**Comment by ankostis** [comment]
It this feature referenced in the docs?

**Pull request merged**